src/plat/imx7/config.cmake

```diff
@@ -9,6 +9,7 @@ cmake_minimum_required(VERSION 3.7.2)
 declare_platform(imx7 KernelPlatImx7 PLAT_IMX7_SABRE KernelSel4ArchAarch32)
 
 if(KernelPlatImx7)
+    config_set(KernelPlatImx7 PLAT_IMX7 ON)
```

Comment:
I think that there was a de-synchronization in the config names here and so PLAT_IMX7 and PLAT_IMX7_SABRE are aliases of each other.  As they have the same CMake cache variable name so cannot ever be set to different values from each other.  It would be nicer to remove one and leave the other but I guess this is hard due to compatibility.

Reply:
I wonder, shouldn't this eventually become something like? Hmmm...didn't we have a similar discussion some time ago already?

```
declare_platform(imx7sabre KernelPlatImx7sabre PLAT_IMX7_SABRE KernelSel4ArchAarch32)
declare_platform(imx7foo KernelPlatImx7foo PLAT_IMX7_FOO KernelSel4ArchAarch32)
declare_platform(imx7bar KernelPlatImx7bar PLAT_IMX7_BAR KernelSel4ArchAarch32)

if(KernelPlatImx7sabre)
    config_set(KernelPlatImx7 PLAT_IMX7 ON)
endif()

if(KernelPlatImx7foo)
    config_set(KernelPlatImx7 PLAT_IMX7 ON)
endif()

  ....

if(KernelPlatImx7)
  ....
```

I've just noticed that the ZYNQ-MP build also does not fully work. Target "zynqmp" can be build but "ultra96" fails due to missing definitions or folder (where it should just use "zynq"). Is there any working sample where one SOC is use on two boards profiles? For imx8 there seem a symlink in place to make this work - but this also feels like a hack to get it to compile

Reply:
Your example is what we talked about some time ago. Generally if the platform has a different device tree description, (due to different hardware resource layout), then it is different enough to be a distinct kernel platform.  This is to avoid ambiguities where a kernel platform name refers to multiple different boards that have different properties that prevent a kernel that is compiled for one variant from correctly running on another.  Essentially, a platform name should be considered an alias for a specific device tree path.

Reply:
I agree with your point here, rooting it in a device tree file (or a base file and some overlays) really makes sense. 

Frankly, I'm not sure how we want to continue now. I don't really insist in having this PR merged, and if this is merged it may set another example of how to not really do it. On the other hand this makes the build work again - which is not worth too much, since nobody seems to be able to test this platform. All I can argue here is, that having a working CMake build at least shows that things do work as they are. All this came from a co-worker asking  me how how adding a new platform is supposed to work and what CMake magic is missing here.

Reply:
I'm happy for this to be merged if it makes the build work again.  I think we should make more of an effort to more clearly mark which patterns are current and which are deprecated. I'm pretty guilty of not doing this.